packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1899,7 +1906,40 @@ export class ProjectService extends BaseService {
         return results;
     }
 
+    async streamResultsToCloudStorage(
+        projectUuid: string,
+        callback: (writer: (data: ResultRow) => void) => Promise<void>,
+    ): Promise<string> {
+        const downloadFileId = nanoid();
+        const passThrough = new PassThrough();
+
+        const endUpload = await this.s3Client.streamResults(
+            passThrough,
+            `${downloadFileId}.jsonl`,
+        );
+
+        const writer = (data: ResultRow) => {
+            passThrough.write(`${JSON.stringify(data)}\n`);
+        };
```

Comment:
should this be inside the try catch?

Reply:
yeah, good point, and also the s3Client.streamResults. so at least we can end the passthrought.